rxfingerprint/src/main/java/com/mtramin/rxfingerprint/FingerprintEncryptionObservable.java

```diff
@@ -115,7 +128,7 @@ protected void onAuthenticationSucceeded(Subscriber<FingerprintEncryptionResult>
 			subscriber.onNext(new FingerprintEncryptionResult(FingerprintResult.AUTHENTICATED, null, encryptedString));
 			subscriber.onCompleted();
 		} catch (CryptoDataException | IllegalBlockSizeException | BadPaddingException | InvalidParameterSpecException | UnsupportedEncodingException e) {
-			subscriber.onError(e);
+			checkKey(e, subscriber, () -> subscriber.onNext(new FingerprintEncryptionResult(FingerprintResult.INVALIDATED, null, null)));
```

Comment:
My thinking here was that if you are encrypting, it could be expected behaviour (hence using `onNext`). However, if decrypting, the data is lost so it should be an error (hence `onError`).

I agree it might be unclear. Do you think from an API standpoint it would be better to just use `FingerprintResult.INVALIDATED` in both cases?